flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NetworkEnvironmentConfiguration.java

```diff
@@ -105,6 +106,302 @@ public NettyConfig nettyConfig() {
+		} else {
+			// use old (deprecated) network buffers parameter
+			int numNetworkBuffers = config.getInteger(TaskManagerOptions.NETWORK_NUM_BUFFERS);
+			networkBufBytes = (long) numNetworkBuffers * (long) segmentSize;
+
+			checkOldNetworkConfig(numNetworkBuffers);
+
+			ConfigurationParserUtils.checkConfigParameter(networkBufBytes < totalJavaMemorySize,
+				networkBufBytes, TaskManagerOptions.NETWORK_NUM_BUFFERS.key(),
+				"Network buffer memory size too large: " + networkBufBytes + " >= " +
+					totalJavaMemorySize + " (total JVM memory size)");
+			ConfigurationParserUtils.checkConfigParameter(networkBufBytes >= segmentSize,
+				networkBufBytes, TaskManagerOptions.NETWORK_NUM_BUFFERS.key(),
+				"Network buffer memory size too small: " + networkBufBytes + " < " +
+					segmentSize + " (" + TaskManagerOptions.MEMORY_SEGMENT_SIZE.key() + ")");
+		}
+
+		return networkBufBytes;
+	}
+
+	/**
+	 * Calculates the amount of memory used for network buffers based on the total memory to use and
+	 * the according configuration parameters.
+	 *
+	 * <p>The following configuration parameters are involved:
+	 * <ul>
+	 *  <li>{@link TaskManagerOptions#NETWORK_BUFFERS_MEMORY_FRACTION},</li>
+	 * 	<li>{@link TaskManagerOptions#NETWORK_BUFFERS_MEMORY_MIN},</li>
+	 * 	<li>{@link TaskManagerOptions#NETWORK_BUFFERS_MEMORY_MAX}</li>
+	 * </ul>.
+	 *
+	 * @param config configuration object
+	 * @param networkBufSize memory of network buffers based on JVM memory size and network fraction
+	 * @param maxJvmHeapMemory maximum memory used for checking the results of network memory
+	 *
+	 * @return memory to use for network buffers (in bytes)
+	 */
+	private static long calculateNewNetworkBufferMemory(Configuration config, long networkBufSize, long maxJvmHeapMemory) {
+		float networkBufFraction = config.getFloat(TaskManagerOptions.NETWORK_BUFFERS_MEMORY_FRACTION);
+		long networkBufMin = MemorySize.parse(config.getString(TaskManagerOptions.NETWORK_BUFFERS_MEMORY_MIN)).getBytes();
+		long networkBufMax = MemorySize.parse(config.getString(TaskManagerOptions.NETWORK_BUFFERS_MEMORY_MAX)).getBytes();
+
+		int pageSize = ConfigurationParserUtils.getPageSize(config);
+
+		checkNewNetworkConfig(pageSize, networkBufFraction, networkBufMin, networkBufMax);
+
+		long networkBufBytes = Math.min(networkBufMax, Math.max(networkBufMin, networkBufSize));
+
+		ConfigurationParserUtils.checkConfigParameter(networkBufBytes < maxJvmHeapMemory,
+			"(" + networkBufFraction + ", " + networkBufMin + ", " + networkBufMax + ")",
+			"(" + TaskManagerOptions.NETWORK_BUFFERS_MEMORY_FRACTION.key() + ", " +
+				TaskManagerOptions.NETWORK_BUFFERS_MEMORY_MIN.key() + ", " +
+				TaskManagerOptions.NETWORK_BUFFERS_MEMORY_MAX.key() + ")",
+			"Network buffer memory size too large: " + networkBufBytes + " >= " +
+				maxJvmHeapMemory + " (maximum JVM memory size)");
+		ConfigurationParserUtils.checkConfigParameter(networkBufBytes >= pageSize,
```

Comment:
this check `networkBufBytes >= pageSize` does not look to happen before for similar code path for `calculateNewNetworkBufferMemory`. Not sure, how important it is, but `networkBufBytes` seems to be calculated in a different way, so the check might be valid for containers but not for network. what do you think?

Reply:
I ever thought that this additional check also makes sense for `calculateNewNetworkBufferMemory`, so reuse together.
You might be right, then I would double check this usage and then confirm with you.

Reply:
I think the check of `networkBufBytes >= pageSize` is redundant in both cases, because the `checkNewNetworkConfig` would verify the `networkBufMin >= pageSize` and `networkBufBytes` should be more than `networkBufMin` based on `Math.min(networkBufMax, Math.max(networkBufMin, networkBufSize))`. Then we could remove that check, what do you think?

Reply:
Agree, looks like it.

Reply:
Was this issue resolved? I still see this check, while from the conversation the consensus was to remove it?

@zhijiangW were there other places where the code's behaviour is different compared to what it was before this refactor?

Reply:
Yes, this issue has already been solved in the commit `[fixup] Fix deprecation and remove unnecessary check`. There are two checks before, and only one check is unnecessary as we confirmed.

I think I have not changed any previous behaviour during refactoring. :)

Reply:
Sorry, my careless made a mistake to only remove check from `NetworkEnvironmentConfiguration#calculateNetworkBufferMemory` and forgot to remove it from `NetworkEnvironmentConfiguration#calculateNewNetworkBufferMemory`.
Thanks for the reminder from Piotr and azagrebin. I already fixed that in a new commit.